### PR TITLE
feat(fsrs): add card rollback to revert a card to its previous state

### LINKED ISSUE #46: Add Rollback method to undo last review
## Problem

No way to undo a review. Users press the wrong button and need to revert the last review.

## Reference

ts-fsrs `fsrs.ts:365-447` — `rollback(card, log)`
```typescript
const card = f.rollback(currentCard, reviewLog)
```

Depends on `ReviewLog.Due` (added in #42).

## Acceptance Criteria

- [ ] `Rollback(card, log)` restores card to state before the review
- [ ] Returns error for manual ratings
- [ ] Results match ts-fsrs for identical inputs

### PR DESCRIPTION
**Pull request by cantalupo555**
## Summary
Adds card rollback functionality to the FSRS package, enabling cards to be reverted to their previous state using review log data. The implementation includes protection against manual ratings and underflow, and aligns with ts-fsrs behavior.

## Motivation
This feature allows users to correct review mistakes or undo accidental reviews by restoring a card's state (due, stability, difficulty, etc.) to its pre-review condition. It ensures consistency with the TypeScript FSRS reference implementation and provides robust error handling for edge cases.

## Changes Made
- Added `ErrManualRating` sentinel error and `Rollback(card Card, log ReviewLog) (Card, error)` method to `fsrs.go`
- Implemented restoration of card fields (State, Due, Stability, Difficulty, ElapsedDays, ScheduledDays, LastReview) from `ReviewLog`
- Added underflow protection: decrements `Reps` only if > 0, and decrements `Lapses` only when rating is Again and `Lapses` > 0
- Preserved `RemainingSteps` from the current card to match ts-fsrs spread behavior
- Added comprehensive test suite `TestRollback` with 12 subtests covering normal operation, edge cases, and ts-fsrs compatibility

## Testing
- [x] Tested locally
- [x] Unit tests added
- [x] No regressions

## Breaking Changes
None

Closes #46

### CODE REVIEW

File: fsrs.go [hunk 1]
```diff
@@ -36,3 +42,45 @@ func (f *FSRS) GetRetrievability(card Card, now time.Time) float64 {
...
+	result.Difficulty = log.Difficulty
+	result.ElapsedDays = log.ElapsedDays
+	result.ScheduledDays = log.ScheduledDays
+	result.LastReview = log.Review
+	if card.Reps > 0 {
+		result.Reps = card.Reps - 1
+	}
+	if log.Rating == Again && card.Lapses > 0 {
+		result.Lapses = card.Lapses - 1
+	}
```
Comment (ishiko732) [inline fsrs.go]: There’s a small issue here: the Lapses count only increments when the card is in the review state and Again is selected. So the condition log.Rating == Again && card.Lapses > 0 isn’t quite correct.

Comment (ishiko732) [reply fsrs.go]: When lapses = 6 and the card’s state is relearning, rating it Again again keeps lapses at 6 rather than increasing it to 7. Therefore, after rollback, lapses should still be 6.

Comment (cantalupo555) [reply fsrs.go]: Good catch!
You're right — `lapses` only increments when the card is in the `Review` state.

I've updated the condition to `log.Rating == Again && log.State == Review` and also aligned the \`Due\`/\`LastReview\`. 

Thanks for the review!

**Review by ishiko732 — APPROVED** [review]
LGTM
